Neos.Neos/Classes/Service/EditorContentStreamZookeeper.php

```diff
@@ -114,28 +75,7 @@ public function relayEditorAuthentication(Authentication\TokenInterface $token):
         if (!$user instanceof User) {
             return;
         }
-        $workspaceName = WorkspaceNameBuilder::fromAccountIdentifier(
-            $token->getAccount()->getAccountIdentifier()
-        );
-        $workspace = $contentRepository->getWorkspaceFinder()->findOneByName($workspaceName);
-        if ($workspace !== null) {
-            return;
-        }
 
-        $baseWorkspace = $contentRepository->getWorkspaceFinder()->findOneByName(WorkspaceName::forLive());
-        if (!$baseWorkspace) {
-            return;
-        }
-        $editorsNewContentStreamId = ContentStreamId::create();
-        $contentRepository->handle(
-            CreateWorkspace::create(
-                $workspaceName,
-                $baseWorkspace->workspaceName,
-                new WorkspaceTitle((string) $user->getName()),
-                new WorkspaceDescription(''),
-                $editorsNewContentStreamId,
-                UserId::fromString($this->persistenceManager->getIdentifierByObject($user))
-            )
-        );
+        $this->workspaceProvider->providePrimaryPersonalWorkspaceForUser($siteDetectionResult->contentRepositoryId, $user);
```

Comment:
i thought the `provide` pattern would be used to instantiate the workspace at the first place actually used. This just eagerly instantiates the workspace - which WILL not be the case for the edge case being logged in already and switching sites - so i would probably prefer to remove the functionality from this service?